src/odemis/driver/simulated.py

```diff
@@ -339,3 +238,195 @@ def __init__(self, name, role, positions, has_pressure=False, **kwargs):
+    """
+    Simulated component, capable of simulating both axes and VA's.
+    This component allows for communication with it, asif it were a real component with axes and VA's, but does not do
+    anything else.
+    """
+    def __init__(self, name, role, vas=None, axes=None, **kwargs):
+        """
+        Only VA's specified in vas are created. Their type is determined based on the supplied initial value, and the
+        presence of range or choices in.
+        Both the presence of VA's and the presence of axes are optional.
+
+        vas (dict (string -> dict (string -> any))): This dict maps desired VA names to a dict with VA properties. The
+        VA property dict can contain the following keys:
+            "value" (any): optional, initial values of the VA, defaults to 0.0
+            "readonly" (bool): optional, True for read only VA, defaults to False
+            "unit" (str): optional, the unit of the VA, defaults to ""
+            "range" (float, float): optional, min/max of the VA, defaults to None
+            "choices" (dict): optional, possible values available to the VA, defaults to None, will be ignored if range
+                              is defined
+        axes (dict (string -> dict (string -> any))): dict mapping desired axis names to dicts with axis properties. The
+        axis property dict can contain the following keys:
+            "unit" (str): optional, unit of the axis, defaults to "m"
+            "range" (float, float): optional, min/max of the axis, defaults to (-0.1, 0.1)
+            "choices" (dict): optional, alternative to ranges, these are the choices of the axis
+            "speed" (float, float): optional, allowable range of speeds, defaults to (0., 10.)
+
+        """
+        # Create desired VA's
+        if vas:
+            for n in vas:
+                if "value" in vas[n]:
+                    vavaluetype = vas[n]["value"].__class__.__name__.capitalize()
+                    if vavaluetype == "Bool":
+                        vavaluetype = "Boolean"
+                    elif vavaluetype == "Str":
+                        vavaluetype = "String"
+                else:
+                    vavaluetype = "VA"
+                if "range" in vas[n]:
+                    varangetype = "Continuous"
+                elif "choices" in vas[n]:
+                    varangetype = "Enumerated"
+                else:
+                    varangetype = "VA"
+                if vavaluetype == "VA" and varangetype == "VA":
+                    vatype = "VigilantAttribute"
+                elif vavaluetype == "VA":  # and varangetype != "VA"
+                    vatype = varangetype + vavaluetype
+                else:
+                    vatype = vavaluetype + varangetype
+
+                if "value" not in vas[n]:
+                    vas[n]["value"] = 0.0  # default value
+                if "unit" not in vas[n]:
+                    vas[n]["unit"] = ""  # default unit
```

Comment:
No, please don't. In VAs we distinguish between unit=None (ie, it'd be meaningless) and unit="" (ie, it's a unit-less value, such as a ratio). Let's keep the standard behaviour which is to have the default unit=None.